### PR TITLE
add metadata about virtual services and destination rules

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

**Comment by rshriram** [comment]
cc @douglas-reid / @kyessenov 
this is what I meant.

**Comment by douglas-reid** [comment]
Thanks for tackling this issue!  This approach seems reasonable to me, though my understanding of the Pilot codebase is limited at best.  It seems like we should be able to extract the attributes via plugin/mixer client.

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -44,7 +44,15 @@ const (
 	PassthroughCluster = "PassthroughCluster"
 	// SniClusterFilter is the name of the sni_cluster envoy filter
 	SniClusterFilter = "envoy.filters.network.sni_cluster"
-
+	// IstioMetadataKey is the key under which metadata is added to a route or cluster
+	// regarding the virtual service or destination rule used for each
+	IstioMetadataKey = "networking.istio.io"
```
Comment (kyessenov) [inline pilot/pkg/networking/util/util.go]: For EDS, this is just "istio". I think we can use "istio" here too (less space).

**Comment by rshriram** [comment]
@kyessenov / @douglas-reid could one of you take over this and get the tests, etc. working?

**Comment by kyessenov** [comment]
Same comment as Mandar: would be nice to capture the changes to xDS output in a test.
A follow up to capture the new metadata in the proxy is also necessary.

**Comment by rshriram** [comment]
This is not adding functionality. Its just for mixer filter to extract info about the virtual service or destination rule being used.. Also the generated config will be very ugly [its proto struct]

**Comment by rshriram** [comment]
> A follow up to capture the new metadata in the proxy is also necessary.
I didn't sign up to augment the mixer filter in proxy to pickup the metadata from the cluster. :). That piece is left to you folks.

**Comment by rshriram** [comment]
Will add tests in a few days but at this moment, this is low priority for me. This was a half hour hack during some meeting, to show you how it can be done. OTOH, if one of you would like to take this PR code and add tests, other embellishments, etc. please feel free to do so

**Comment by kyessenov** [comment]
@rshriram Thanks for the change. Yes, it's' up to telemetry team to update proxy to take advantage of the fields. But it would be good to have tests that cover this feature, as a form of contract on the part of envoy config generation.

**Pull request closed without merging**